src/ngraph/op/quantized_convolution.cpp

```diff
@@ -0,0 +1,196 @@
+                                               const shared_ptr<Node>& output_scale,
+                                               const shared_ptr<Node>& output_zero_point,
+                                               const element::Type& output_type,
+                                               const AxisSet& input_axes,
+                                               const AxisSet& filter_axes,
+                                               const AxisSet& output_axes)
+    : Op("QuantizedConvolution",
+         check_single_output_args({input,
+                                   filters,
+                                   input_scale,
+                                   input_zero_point,
+                                   filter_scale,
+                                   filter_zero_point,
+                                   output_scale,
+                                   output_zero_point}))
+    , m_window_movement_strides(window_movement_strides)
+    , m_window_dilation_strides(window_dilation_strides)
+    , m_padding_below(padding_below)
+    , m_padding_above(padding_above)
+    , m_data_dilation_strides(data_dilation_strides)
+    , m_output_type(output_type)
+    , m_input_axes(input_axes)
+    , m_filter_axes(filter_axes)
+    , m_output_axes(output_axes)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::QuantizedConvolution::validate_and_infer_types()
+{
+    enum
+    {
+        INPUT_SCALE = 2,
+        INPUT_ZERO_POINT,
+        FILTER_SCALE,
+        FILTER_ZERO_POINT,
+        OUTPUT_SCALE,
+        OUTPUT_ZERO_POINT
+    };
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(INPUT_SCALE).is_real() ||
+                              get_input_element_type(FILTER_SCALE).is_real() ||
+                              get_input_element_type(OUTPUT_SCALE).is_real(),
+                          "Scale must be a floating point number");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(0) == get_input_element_type(INPUT_ZERO_POINT),
+                          "Input Zero point element type (",
+                          get_input_element_type(INPUT_ZERO_POINT),
+                          ") must match input element type (",
+                          get_input_element_type(0),
+                          ")");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(1) == get_input_element_type(FILTER_ZERO_POINT),
+                          "Filter Zero point element type (",
+                          get_input_element_type(FILTER_ZERO_POINT),
+                          ") must match filter element type (",
+                          get_input_element_type(1),
+                          ")");
+
+    // TODO Remove these checks once we support channelwise and vector of scales
+    NODE_VALIDATION_CHECK(this,
+                          shape_size(get_input_shape(2)) == 1 ||
```

Comment:
(This one will be a bit of a pain because a needed utility function is not defined yet.)

We'll need to define a `shape_size`-like function that works on `PartialShape`s. This should have a return type of `Dimension`, and the expression `partial_shape_size(pshape)` should return:

(1) `Dimension::dynamic()` if the rank of `pshape` PartialShape is dynamic
(2) `Dimension::dynamic()` if any dimension of `pshape` is unknown (except, I suppose, it could return 0 if there is a 0 anywhere)
(3) The same value as `shape_size(pshape.to_shape())`, if `pshape.is_static()` is true.

Then, this check would become `partial_shape_size(get_input_partial_shape(2)).compatible(1)`.

Reply:
Similar for all `shape_size` checks in the next few lines here.

Reply:
I'm a little confused about this check, actually... why is it `||`?